### PR TITLE
42 - Form: Starting point

### LINKED ISSUE #42: [5] Form: Starting Point
Our multi step form requires several unique sections, each section has its own design and data challenges.

This part of the form captures the starting point, it is used to capture the starting point of our journey.

This ticket is not interested in performing the lookups eventually required to find the exact starting point of a journey

## Acceptance Criteria

- [x] A single text input box with the label "Where's your journey starting?"
- [x] A back link that will take you back to the homepage
- [x] Entering text into the fiend will present you with multiple potential options (these can be hard coded as part of this ticket)

### LINKED ISSUE #89: Setup Selenium webdriver to test features with JavaScript
In order to run feature tests that use JavaScript for example the autocomplete field we need to run a simulated browser for our tests to work.

Without it, we can't test code that depends on JavaScript and our tests will break.

Read more about that here
https://github.com/teamcapybara/capybara#drivers

## Acceptance criteria

- [x] We have a driver installed that can runs tests with JavaScript
- [x] It works locally
- [x] It works in our CI/CD environment

### PR DESCRIPTION
**Pull request by francosta**
Apologies @tosbourn and @mathewtrivett for the amount of commits and file changes in a single PR. As I said, this has largely to do with the need to setup several things to get started on the current issue.

Things that are included in here but are not strictly related to the issue:
- Added `rails-controller-testing` gem. This is something I should have added when setting up the testing framework;
- Added the `wicked` gem to support a multi-step form;
- Created a `Trip` model with a `starting_point` `string` attribute, with validation for `presence`;
- Added a controller for `Trip` with a single `route` for `/trips/new`. Added logic to this action to redirect to the `trip_builder` controller;
- Added validation test for `starting_point  on `Trip` model;
- Added test for `Trip` controller;

Things that are related to the issue:
- Added a controller for the TripBuilder in order to support the form with `wicked`. This controller only has a `starting_point` `:step` for now. Also included the minimum setup for the wizard to render and function;
- Installed the `js-autocomplete` npm package to support autocomplete functionality on `starting_point` input form.
- Added autocomplete logic in `javascript/components/autocomplete.js` with hardcoded options for dropdown and no call to the Google Places API;
- Added view for `starting_point`;
- Added internationalization variables for `starting_point` form and link;
- Added test for `show` on `TripBuilderController`;

I will need some help with tests. I've done the most I could but have not been able to test the view created for `starting_point` and, possibly, others. I don't think we should want to merge this before at least there are some assertions about what's in the view.

Closes #42 

## Notes from Mat

This PR also closes #89. Some of our tests depend on JavaScript. To make these tests run we need to have a browser driver installed anywhere we want to run tests. There instructions on how to do this in the README on this branch, the CI pipeline is updated to make sure that browser drivers are installed.

### CODE REVIEW

**Review by mathewtrivett — COMMENTED** [review]
This looks like a great start Fran. Let's pick up pairing on writing tests for the views together tomorrow.

File: app/views/trip_builder/starting_point.html.erb [hunk 1]
```diff
@@ -1,11 +1,10 @@
-<%= link_to "Back", "/"%>
+<%= link_to "#{t('back')}", "/"%>
```
[the remark this answers is not in the document]
Comment (francosta) [reply app/views/trip_builder/starting_point.html.erb:1]: Yes Mat - thanks. I definitely agree on the handlers and will change it. 🙌

[the remark this answers is not in the document]
Comment (tosbourn) [reply app/views/trip_builder/starting_point.html.erb:1]: One small comment about this line as well.

Anytime we write `"#{something}"` and there are no other bits of the string, we can remove the need to interpolate that data by just writing `something`.

In this case:

``` diff
- <%= link_to "#{t('back')}", "/"%>
+ <%= link_to t('back'), root_path %>
```

File: spec/test/controllers/trip_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe TripsController do
+  describe '#new', type: :controller do
+    it 'will redirect to the trip builder' do
+      get :new
+      response.status.should be(302)
+      response.should redirect_to '/trip_builder'
```
[the remark this answers is not in the document]
Comment (tosbourn) [reply spec/test/controllers/trip_controller_spec.rb:8]: ideally, yes, we would use a URL helper here.

The rule of thumb should be:

- If we care about the specifics of the URL for branding reasons, compare against a hardcoded string
- Any other time, use a helper

For example, if the test was "we always use the term articles for the URL, even though our system calls them posts" (terrible test name, but you get the idea) then we would care that `/articles` was hit.

**Review by tosbourn — CHANGES_REQUESTED** [review]
A few minor changes, but looking great. I love how much you've accomplished with relatively little code.

**Comment by tosbourn** [comment]
Also, thanks for the super clear PR description, so easy to read!

**Pull request merged**